Engine/Alphas/ChartingInsightManagerExtension.cs

```diff
@@ -88,19 +86,31 @@ public ChartingInsightManagerExtension(IAlgorithm algorithm, StatisticsInsightMa
         /// <param name="frontierTimeUtc">The current frontier time utc</param>
         public void Step(DateTime frontierTimeUtc)
         {
+            // Only add our charts to the algorithm when we actually have an insight 
+            // We will still update our internal charts anyways, but this keeps Alpha charts out of
+            // algorithms that don't use the framework.
+            if (!_chartsAdded && _dailyInsightCount > 0)
+            {
+                _algorithm.AddChart(_insightScoreChart);
+                _algorithm.AddChart(_totalInsightCountChart);
+                _algorithm.AddChart(_totalInsightCountPerSymbolChart);
+
+                _chartsAdded = true;
+            }
+
             // sample insight/symbol counts each utc day change
             if (frontierTimeUtc.Date > _lastInsightCountSampleDateUtc)
             {
                 _lastInsightCountSampleDateUtc = frontierTimeUtc.Date;
 
                 // add sum of daily insight counts to the total insight count series
-                _totalInsightCountSeries.AddPoint(frontierTimeUtc.Date, _dailyCount);
+                _totalInsightCountChart.Series["Count"].AddPoint(frontierTimeUtc.Date, _dailyInsightCount);
```

Comment:
`_totalInsightCountSeries` was a direct access vs `.Series["Count"].` which causes a lookup on each call?

Reply:
Thats true, I'll switch back and keep the series as a class var 👍🏽